buffer/src/main/java/io/netty/buffer/ByteBufAllocatorMetricProvider.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer;
+
+public interface ByteBufAllocatorMetricProvider {
```

Comment:
@Scottmitch which confuses me... as if we deprecate them on the allocator and so tell the user to use the metric interface it would be more clear, or I am missing anything ?